clients/da-vinci-client/src/main/java/com/linkedin/davinci/consumer/VeniceChangelogConsumerImpl.java

```diff
@@ -153,9 +168,40 @@ public VeniceChangelogConsumerImpl(
     if (changelogClientConfig.getViewName() == null || changelogClientConfig.getViewName().isEmpty()
         || changelogClientConfig.isBeforeImageView()) {
       this.storeName = changelogClientConfig.getStoreName();
+      rocksDBStorageEngineFactory = null;
+      // The in memory storage engine only relies on the name of store and nothing else. We use an unversioned store
+      // name
+      // here in order to reduce confusion (as this storage engine can be used across version topics).
+      InMemoryStorageEngine inMemoryStorageEngine = new InMemoryStorageEngine(storeName);
+      this.chunkAssembler = new InMemoryChunkAssembler(inMemoryStorageEngine);
     } else {
       this.storeName =
           VeniceView.getViewStoreName(changelogClientConfig.getStoreName(), changelogClientConfig.getViewName());
+      Properties rocksDBBufferProperties = new Properties();
+      String rocksDBBufferPath = changelogClientConfig.getBootstrapFileSystemPath();
+      if (rocksDBBufferPath == null || rocksDBBufferPath.isEmpty()) {
+        throw new VeniceException("bootstrapFileSystemPath must be configured for consuming view store: " + storeName);
+      }
+      // Create a new folder in user provided path so if the path contains other important files we don't drop it.
+      rocksDBBufferProperties
```

Comment:
Question:
Do we expect this folder carry the previous state after a rolling bounce?
If yes, we need to flush the storage engine at checkpointing time, and if no, we can potentially clean up all the previous files at startup time, right?

Reply:
Good question, the change log consumer's checkpointing is interesting. The `seekToCheckpoint` API takes in user provided `Set<VeniceChangeCoordinate> checkpoints`. IIUC this means we can't guarantee not dropping records unless we flush after every poll. Otherwise even if the user seek to the exact spot after restart we could be missing chunks during assembly and we simply catch all errors and log:
```
LOGGER.warn(
            "Encountered error assembling chunked record, this can happen when seeking between chunked records. Skipping offset {} on topic {}",
            recordOffset,
            pubSubTopicPartition.getPubSubTopic().getName());
```